### PR TITLE
WIP: added hxsurf addition

### PR DESCRIPTION
**Pull request by dokato**
I needed an easy way to concatenate 2 hxsurfs, I did not find any better way to do it, so I implemented this simple method. For example:

```r
h1 = as.hxsurf(icosahedron3d(), 'a')
h2 = as.hxsurf(tetrahedron3d(), 'b')
h3=h1+h2
plot3d(h3)
```
Screenshot 2021-09-01 at 15 20 38

The problem is that right now hxsurf inherits operations from dotprops, which makes sense for most but not all of them.

```r
Ops.hxsurf <- Ops.dotprops
```

I'm not sure how to point the package to what is the desired method to call in that case.

### CODE REVIEW

**Comment by jefferis** [comment]
Thanks a lot for this @dokato. I think this is where S3 methods are not so good as really the method to choose here depends on both arguments i.e. whether to do arithmetic or join things together. I would actually sidestep this by using `c` i.e. `c.hxsurf` or `merge` rather than `+` for this functinolality.

File: R/hxsurf.R [hunk 1]
```diff
@@ -770,8 +770,8 @@ contains_points.ashape3d<-function(obj, points,  ...) {
 #' @examples
 #' h1 = as.hxsurf(icosahedron3d(), 'a')
 #' h2 = as.hxsurf(tetrahedron3d(), 'b')
-#' h3=h1+h2
-`+.hxsurf` <- function(x, y) {
+#' h3=c(h1, h2)
+`c.hxsurf` <- function(x, y) {
```
Comment (jefferis) [inline R/hxsurf.R:774]: Thanks! But doesn't the signature needs to be `c(...)` to match base `c`. I would expect R CMD check to complain ...

Comment (dokato) [reply R/hxsurf.R:774]: yes, working on that right now :)

Comment (dokato) [reply R/hxsurf.R]: PTAL now

File: R/hxsurf.R [hunk 2]
```diff
@@ -759,3 +759,48 @@ contains_points.hxsurf<-contains_points.mesh3d
...
+
+#' Concatenate two HyperSurface objects
+#'
+#' @param x hxsurf
+#' @param y another hxsurf
+#' @return new hxsurf
+#' @examples
+#' h1 = as.hxsurf(icosahedron3d(), 'a')
+#' h2 = as.hxsurf(tetrahedron3d()+1, 'b')
+#' h3=c(h1, h2)
```
Comment (jefferis) [inline R/hxsurf.R:771]: I forget - will this appear in the docs (although function is not exported)?

**Review by jefferis — APPROVED** [review]
Looks great. Thank you!

File: R/hxsurf.R (hunk 2 above)
Comment (jefferis) [reply R/hxsurf.R:771]: I think it does. I would not document publicly since not exported. Maybe just change `#'` to `#`?

Comment (dokato) [reply R/hxsurf.R]: True, I find docs of internal functions useful, esp. when I need to use `:::`, but I know that this is an "unofficial" scenario. Fixed.

**Pull request merged**